server/controllers/caseNotesController.ts

```diff
@@ -288,14 +288,22 @@ export default class CaseNotesController {
   }
 
   public postUpdate(): RequestHandler {
-    return async (req: Request, res: Response) => {
+    return async (req: Request, res: Response, next: NextFunction) => {
       const { prisonerNumber, caseNoteId } = req.params
       const { text, isExternal, currentLength, username } = req.body
+      const { clientToken } = res.locals
+      const currentCaseNote = await this.caseNotesService.getCaseNote(clientToken, prisonerNumber, caseNoteId)
+      const canEditSensitiveCaseNotes = userHasRoles([Role.PomUser], res.locals.user.userRoles)
+
+      if (currentCaseNote.sensitive && !canEditSensitiveCaseNotes) {
+        logger.info(`User not permitted to edit sensitive case notes`)
+        return next()
+      }
```

Comment:
When displaying the update page, do a check to ensure that a sensitive case note can be accessed.